instrumentation-annotations-support/src/test/java/io/opentelemetry/instrumentation/api/annotation/support/ParameterizedClassTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.annotation.support;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.Serializable;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.RandomAccess;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.jupiter.api.Test;
+
+class ParameterizedClassTest {
+
+  Type stringArrayListType = getGeneric(TestFields.class, "stringArrayListField");
+
+  static Type getGeneric(Class<?> clazz, String fieldName) {
+    try {
+      return clazz.getField(fieldName).getGenericType();
+    } catch (NoSuchFieldException e) {
+      fail("No such field");
+      throw new RuntimeException("No such field", e);
+    }
+  }
+
+  @Test
+  void reflectsClass() {
+
+    ParameterizedClass underTest = ParameterizedClass.of(ArrayList.class);
+    assertThat(underTest.getRawClass()).isEqualTo(ArrayList.class);
+    assertThat(underTest.getActualTypeArguments())
+        .hasSize(1)
+        .allMatch(x -> x instanceof TypeVariable);
+  }
+
+  @Test
+  void reflectsParameterizedType() {
+    ParameterizedClass underTest = ParameterizedClass.of(stringArrayListType);
+    assertThat(underTest.getRawClass()).isEqualTo(ArrayList.class);
+    assertThat(underTest.getActualTypeArguments()).hasSize(1).allMatch(x -> x == String.class);
+  }
+
+  @Test
+  void getsParameterizedSuperclassWithMappedtypeArguments() {
+    ParameterizedClass underTest =
+        ParameterizedClass.of(stringArrayListType).getParameterizedSuperclass();
+    assertThat(underTest.getActualTypeArguments()).hasSize(1).allMatch(x -> x == String.class);
```

Comment:
Wouldn't
```suggestion
    assertThat(underTest.getActualTypeArguments()).containsExactly(String.class);
```
work as well?

Reply:
yeah, looks like it. That's me not knowing assert4j api surface well enough. Will change it (2 places).